### ISSUE TITLE
Build errors through failure

### ISSUE THREAD

**Issue by mpapierski** [alexcrichton/backtrace-rs#134]
Apologizes if this is incorrect repo to report such things. Probably I should report in both affected repos.

I'm using `failure = 0.1.3` and after `cargo update` backtrace got resolved to `0.3.10` which is released today. And all of sudden I can't build failure because:

```
   Compiling failure v0.1.3                                                                                                                                                                                        
error[E0034]: multiple applicable items in scope                                                                                                                                                                   
   --> /home/michal/.cargo/registry/src/github.com-1ecc6299db9ec823/failure-0.1.3/src/backtrace/mod.rs:132:20                                                                                                      
    |                                                                                                                                                                                                              
132 |                 bt.fmt(f)                                                                                                                                                                                    
    |                    ^^^ multiple `fmt` found                                                                                                                                                                  
    |                                                                                                                                                                                                              
    = note: candidate #1 is defined in an impl of the trait `std::fmt::Debug` for the type `backtrace::backtrace::Backtrace`                                                                                       
    = note: candidate #2 is defined in an impl of the trait `std::fmt::Display` for the type `backtrace::backtrace::Backtrace`                                                                                     
                                                                                                                                         
```

Looks like backtrace introduced some backwards incompatible change, or failure needs to be updated to accomodate for changes in backtrace.

For now `cargo update -p backtrace --precise 0.3.9` solves the build issue, but as I'm developing a lib crate my travis builds will fail.

**Comment by atoav** [comment]
Could reproduce the same issue in a minimal environment (the default `cargo new`-
_hello world_ with `extern crate failure` and `failure="0.1.3"` in `Cargo.toml`)

**Comment by aleksuss** [comment]
FYI https://github.com/rust-lang-nursery/failure/pull/279

**Comment by vlisivka** [comment]
More errors after fix in 0.3.11:

```
   Compiling backtrace v0.3.11                                                                                                                    
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:156:60                       
    |                                                                                                                                             
156 |         STATE = bt::backtrace_create_state(ptr::null(), 0, error_cb,                                                                        
    |                                                            ^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`            
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, *const u8, i32)`                                                                 
               found type `extern "C" fn(*mut libc::c_void, *const u8, i32) {symbolize::libbacktrace::error_cb}`                                  
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:156:60                       
    |                                                                                                                                             
156 |         STATE = bt::backtrace_create_state(ptr::null(), 0, error_cb,                                                                        
    |                                                            ^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`            
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, *const i8, i32)`                                                                 
               found type `extern "C" fn(*mut libc::c_void, *const i8, i32) {symbolize::libbacktrace::error_cb}`                                  
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:172:36                       
    |                                                                                                                                             
172 |                                    pcinfo_cb, error_cb,                                                                                     
    |                                    ^^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`                                   
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, usize, *const u8, i32, *const u8) -> i32`                                        
               found type `extern "C" fn(*mut libc::c_void, usize, *const u8, i32, *const u8) -> i32 {symbolize::libbacktrace::pcinfo_cb}`        
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:172:36                       
    |                                                                                                                                             
172 |                                    pcinfo_cb, error_cb,                                                                                     
    |                                    ^^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`                                   
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, usize, *const i8, i32, *const i8) -> i32`                                        
               found type `extern "C" fn(*mut libc::c_void, usize, *const i8, i32, *const i8) -> i32 {symbolize::libbacktrace::pcinfo_cb}`        
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:172:47                       
    |                                                                                                                                             
172 |                                    pcinfo_cb, error_cb,                                                                                     
    |                                               ^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`                         
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, *const u8, i32)`                                                                 
               found type `extern "C" fn(*mut libc::c_void, *const u8, i32) {symbolize::libbacktrace::error_cb}`                                  
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:172:47                       
    |                                                                                                                                             
172 |                                    pcinfo_cb, error_cb,                                                                                     
    |                                               ^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`                         
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, *const i8, i32)`                                                                 
               found type `extern "C" fn(*mut libc::c_void, *const i8, i32) {symbolize::libbacktrace::error_cb}`                                  
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:176:31                       
    |                                                                                                                                             
176 |                               syminfo_cb, error_cb,                                                                                         
    |                               ^^^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`                                       
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, usize, *const u8, usize, usize)`                                                 
               found type `extern "C" fn(*mut libc::c_void, usize, *const u8, usize, usize) {symbolize::libbacktrace::syminfo_cb}`                
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:176:31                       
    |                                                                                                                                             
176 |                               syminfo_cb, error_cb,                                                                                         
    |                               ^^^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`                                       
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, usize, *const i8, usize, usize)`                                                 
               found type `extern "C" fn(*mut libc::c_void, usize, *const i8, usize, usize) {symbolize::libbacktrace::syminfo_cb}`                
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:176:43                       
    |                                                                                                                                             
176 |                               syminfo_cb, error_cb,                                                                                         
    |                                           ^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`                             
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, *const u8, i32)`                                                                 
               found type `extern "C" fn(*mut libc::c_void, *const u8, i32) {symbolize::libbacktrace::error_cb}`                                  
                                                                                                                                                  
error[E0308]: mismatched types                                                                                                                    
   --> /home/vlisivka/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.11/src/symbolize/libbacktrace.rs:176:43                       
    |                                                                                                                                             
176 |                               syminfo_cb, error_cb,                                                                                         
    |                                           ^^^^^^^^ expected enum `core::ffi::c_void`, found enum `libc::c_void`                             
    |                                                                                                                                             
    = note: expected type `extern "C" fn(*mut core::ffi::c_void, *const i8, i32)`                                                                 
               found type `extern "C" fn(*mut libc::c_void, *const i8, i32) {symbolize::libbacktrace::error_cb}`                                  
                                                                                                                                                  
error: aborting due to 5 previous errors                                                                                                          

```

**Comment by vlisivka** [comment]
Version 0.3.9 works fine. Use command `cargo update -p backtrace --precise 0.3.9` as workaround.

**Comment by alexcrichton** [comment]
I'm going to close this since I believe it was fixed with 0.3.11

@vlisivka are you perhaps using an outdated rustc? If not, mind opening a dedicated issue with some more info?
